test/core/gc.spec.js

```diff
@@ -0,0 +1,245 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const isNode = require('detect-node')
+const pEvent = require('p-event')
+const IPFS = require('../../src/core')
+const createTempRepo = require('../utils/create-repo-nodejs')
+
+describe('gc', function () {
+  const fixtures = [{
+    path: 'test/my/path1',
+    content: Buffer.from('path1')
+  }, {
+    path: 'test/my/path2',
+    content: Buffer.from('path2')
+  }, {
+    path: 'test/my/path3',
+    content: Buffer.from('path3')
+  }, {
+    path: 'test/my/path4',
+    content: Buffer.from('path4')
+  }]
+
+  let ipfs
+  let repo
+
+  before(function (done) {
+    this.timeout(20 * 1000)
+    repo = createTempRepo()
+    let config = { Bootstrap: [] }
+    if (isNode) {
+      config.Addresses = {
+        Swarm: ['/ip4/127.0.0.1/tcp/0']
+      }
+    }
+    ipfs = new IPFS({ repo, config })
+    ipfs.on('ready', done)
+  })
+
+  after(function (done) {
+    this.timeout(60 * 1000)
+    ipfs.stop(done)
+  })
+
+  after((done) => repo.teardown(done))
+
+  const blockAddTests = [{
+    name: 'add',
+    add1: () => ipfs.add(fixtures[0], { pin: false }),
+    add2: () => ipfs.add(fixtures[1], { pin: false }),
+    resToCid: (res) => res[0].hash
+  }, {
+    name: 'object put',
+    add1: () => ipfs.object.put({ Data: 'obj put 1', Links: [] }),
+    add2: () => ipfs.object.put({ Data: 'obj put 2', Links: [] }),
+    resToCid: (res) => res.toString()
+  }, {
+    name: 'block put',
+    add1: () => ipfs.block.put(Buffer.from('block put 1'), null),
+    add2: () => ipfs.block.put(Buffer.from('block put 2'), null),
+    resToCid: (res) => res.cid.toString()
+  }]
+
+  describe('locks', function () {
```

Comment:
all theses tests are specific to js-ipfs ? or can be interface-core tests ?

Reply:
These tests are quite specific and need to be able to control timing using events / hooks, so I think it makes most sense for them to be here